src/main/scala/pureconfig/ConfigFieldMapping.scala

```diff
@@ -1,24 +1,48 @@
 package pureconfig
 
-trait ConfigFieldMapping[T] {
-  def toConfigField(fieldName: String): String
-}
-
-class IdentityConfigFieldMapping[T] extends ConfigFieldMapping[T] {
-  def toConfigField(fieldName: String): String = fieldName
-}
-
-class WordDelimiterConfigFieldMapping[T](
-    typeFieldDelimiter: WordDelimiter,
-    configFieldDelimiter: WordDelimiter) extends ConfigFieldMapping[T] {
-  def toConfigField(fieldName: String): String =
-    (typeFieldDelimiter.toTokens _ andThen configFieldDelimiter.fromTokens _)(fieldName)
+trait ConfigFieldMapping[T] extends (String => String) {
+  def apply(fieldName: String): String
 }
 
 object ConfigFieldMapping extends LowPriorityConfigFieldMappingImplicits {
   def apply[T](implicit mapping: ConfigFieldMapping[T]): ConfigFieldMapping[T] = mapping
```

Comment:
Is this needed? If so is it worth documenting why it exists? (Not a rhetorical question; I really don't know.)